### PR TITLE
[Horizon] Inbox

### PR DESCRIPTION
**Pull request by Ahmed-Naguib93**
Add Inbox feature like student app.

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>You have included an invalid app. Valid values are: Student, Teacher, Parent, None</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Added inbox
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 45503255b01ce6660fa6a5a1546591e7027a2eb6
</p>

**Review by szabinst — CHANGES_REQUESTED** [review]
Please add `// TODO: xyz` to all the color changes so we don't forget to change it back once we have high fidelity designs.

File: Horizon/Horizon/Sources/Features/HorizonTabBarController.swift [hunk 1]
```diff
@@ -68,4 +69,22 @@ class HorizonTabBarController: UITabBarController, UITabBarControllerDelegate {
...
+
+        inboxController = InboxAssembly.makeInboxViewController()
+
+        let empty = CoreNavigationController()
+
+        inboxSplit.viewControllers = [inboxController, empty]
+        let title = "Inbox"
+        inboxSplit.tabBarItem = UITabBarItem(title: title, image: .inboxTab, selectedImage: .inboxTabActive)
+        inboxSplit.tabBarItem.accessibilityIdentifier = "TabBar.inboxTab"
+        inboxSplit.tabBarItem.makeUnavailableInOfflineMode()
```
Comment (szabinst) [inline Horizon/Horizon/Sources/Features/HorizonTabBarController.swift:85]: Please comment out this for now as we don't have exact plans for offline support.

File: Horizon/Horizon/Sources/Routing/HorizonRouter.swift [hunk 2]
```diff
@@ -61,3 +83,59 @@ private func pageViewController(url: URLComponents, params: [String: String], us
     }
     return PageDetailsViewController.create(context: context, pageURL: pageURL, app: .student)
 }
+
+private func offlineFileDetails(
```
Comment (szabinst) [inline Horizon/Horizon/Sources/Routing/HorizonRouter.swift:87]: We don't need this for now.

File: Horizon/Horizon/Sources/Routing/HorizonRouter.swift [hunk 3]
```diff
@@ -45,7 +45,29 @@ enum HorizonRouter {
...
+                    return true
+                }
+            }()
+            return MessageDetailsAssembly.makeViewController(
+                env: AppEnvironment.shared,
+                conversationID: conversationID,
+                allowArchive: allowArchive
+            )
+        },
+        RouteHandler("/courses/:courseID/files/:section/:resourceID/:fileID/offline", factory: offlineFileDetails),
```
Comment (szabinst) [inline Horizon/Horizon/Sources/Routing/HorizonRouter.swift:64]: We don't need this for now.

**Comment by inst-danger** [comment]
<details><summary>Horizon Build QR Code:</summary>

</details>

**Review by szabinst — APPROVED** [review]
QA + 1

**Pull request merged**